tests/unit/test_paragraphHelper.py

```diff
@@ -0,0 +1,207 @@
+		self.assertEqual(ti.text, "Paragraph 3.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(False, ti)  # second paragraph
+		self.assertTrue(moved)
+		ti.expand(textInfos.UNIT_LINE)
+		self.assertEqual(ti.text, "Paragraph 2.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(False, ti)  # first paragraph
+		self.assertTrue(moved)
+		ti.expand(textInfos.UNIT_LINE)
+		self.assertEqual(ti.text, "Paragraph 1.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(False, ti)  # no more paragraphs
+		self.assertFalse(moved)
+
+	def test_multiLineBreakParagraph_forward(self):
+		"""
+			Along with double line break paragraphs (block style),
+			the algorithm supports any number of blank lines between block paragraphs.
+			It also supports white space in either the blank lines,
+			or preceeding the block paragraph itself.
+		"""
+		text = """
+
+
+Paragraph 1.
+
+
+\tParagraph 2.
+
+
+
+  \t\t\t
+
+
+     Paragraph 3.
+
+
+
+		"""
+		obj = BasicTextProvider(text=text)
+		ti = obj.makeTextInfo(textInfos.POSITION_FIRST)
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(True, ti)  # first paragraph
+		self.assertTrue(moved)
+		ti.expand(textInfos.UNIT_LINE)
+		self.assertEqual(ti.text, "Paragraph 1.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(True, ti)  # second paragraph
+		self.assertTrue(moved)
+		ti.expand(textInfos.UNIT_LINE)
+		self.assertEqual(ti.text, "\tParagraph 2.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(True, ti)  # third paragraph
+		self.assertTrue(moved)
+		ti.expand(textInfos.UNIT_LINE)
+		self.assertEqual(ti.text, "     Paragraph 3.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(True, ti)  # no more paragraphs
+		self.assertFalse(moved)
+
+	def test_multiLineBreakParagraph_backward(self):
+		"""
+			Test moving back when paragraphs are separated by multiple blank lines / whitespace
+		"""
+		text = """
+
+
+Paragraph 1.
+
+
+\tParagraph 2.
+
+
+
+  \t\t\t
+
+
+     Paragraph 3.
+
+
+
+		"""
+		obj = BasicTextProvider(text=text)
+		ti = obj.makeTextInfo(textInfos.POSITION_LAST)
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(False, ti)  # third paragraph
+		self.assertTrue(moved)
+		ti.expand(textInfos.UNIT_LINE)
+		self.assertEqual(ti.text, "     Paragraph 3.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(False, ti)  # second paragraph
+		self.assertTrue(moved)
+		ti.expand(textInfos.UNIT_LINE)
+		self.assertEqual(ti.text, "\tParagraph 2.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(False, ti)  # first paragraph
+		self.assertTrue(moved)
+		ti.expand(textInfos.UNIT_LINE)
+		self.assertEqual(ti.text, "Paragraph 1.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(False, ti)  # no more paragraphs
+		self.assertFalse(moved)
+
+	def test_DoubleLineBreakWithSingleLineBreakContent(self):
+		"""
+			Algorithm should ignore single line breaks when moving to content with two ore more line breaks.
```

Comment:
Indentation and typo fix
```suggestion
		Algorithm should ignore single line breaks when moving to content with two or more line breaks.
```